lib/gate.h

```diff
@@ -37,43 +58,82 @@ struct Gate {
 
   GateKind kind;
   unsigned time;
-  unsigned num_qubits;
   std::vector<unsigned> qubits;
+  std::vector<unsigned> controlled_by;
+  uint64_t cmask;
   std::vector<fp_type> params;
   Matrix<fp_type> matrix;
-  bool unfusible;      // If true, the gate is fused as a master.
-  bool inverse;        // If true, the qubit order is inversed (q0 > q1).
+  bool unfusible;      // If true, the gate is fused as a parent.
+  bool swapped;        // If true, the qubit order is inversed (q0 > q1).
```

Comment:
Does this still indicate inverted qubit order, or is another ordering (e.g. `0, 2, 1` -> `0, 1, 2` for a controlled gate) possible? Please update the comment if so.

Reply:
Done.